archive/terraform/api_gw_lambda/api_gateway.tf

```diff
@@ -1,47 +1,46 @@
-resource "aws_api_gateway_rest_api" "archive_asset_lookup" {
-  name        = "archive_asset_lookup"
-  description = ""
+resource "aws_api_gateway_rest_api" "api" {
+  name        = "${var.name}"
+  description = "${var.description}"
 }
 
 resource "aws_api_gateway_resource" "proxy" {
-  rest_api_id = "${aws_api_gateway_rest_api.archive_asset_lookup.id}"
-  parent_id   = "${aws_api_gateway_rest_api.archive_asset_lookup.root_resource_id}"
+  rest_api_id = "${aws_api_gateway_rest_api.api.id}"
+  parent_id   = "${aws_api_gateway_rest_api.api.root_resource_id}"
   path_part   = "{proxy+}"
 }
 
 resource "aws_api_gateway_method" "proxy" {
-  rest_api_id   = "${aws_api_gateway_rest_api.archive_asset_lookup.id}"
+  rest_api_id   = "${aws_api_gateway_rest_api.api.id}"
   resource_id   = "${aws_api_gateway_resource.proxy.id}"
   http_method   = "ANY"
-  authorization = "CUSTOM"
-  authorizer_id = "NONE"
+  authorization = "NONE"
 }
 
 resource "aws_api_gateway_integration" "lambda" {
-  rest_api_id = "${aws_api_gateway_rest_api.archive_asset_lookup.id}"
+  rest_api_id = "${aws_api_gateway_rest_api.api.id}"
   resource_id = "${aws_api_gateway_resource.proxy.id}"
   http_method = "${aws_api_gateway_method.proxy.http_method}"
 
   integration_http_method = "POST"
   type                    = "AWS_PROXY"
-  uri                     = "${module.lambda_archive_asset_lookup.arn}"
+  uri                     = "${module.lambda.invoke_arn}"
 }
 
 resource "aws_api_gateway_method" "proxy_root" {
-  rest_api_id   = "${aws_api_gateway_rest_api.archive_asset_lookup.id}"
-  resource_id   = "${aws_api_gateway_rest_api.archive_asset_lookup.root_resource_id}"
+  rest_api_id   = "${aws_api_gateway_rest_api.api.id}"
+  resource_id   = "${aws_api_gateway_rest_api.api.root_resource_id}"
   http_method   = "ANY"
   authorization = "NONE"
 }
 
 resource "aws_api_gateway_integration" "lambda_root" {
-  rest_api_id = "${aws_api_gateway_rest_api.archive_asset_lookup.id}"
+  rest_api_id = "${aws_api_gateway_rest_api.api.id}"
   resource_id = "${aws_api_gateway_method.proxy_root.resource_id}"
   http_method = "${aws_api_gateway_method.proxy_root.http_method}"
 
   integration_http_method = "POST"
   type                    = "AWS_PROXY"
-  uri                     = "${module.lambda_archive_asset_lookup.invoke_arn}"
+  uri                     = "${module.lambda.invoke_arn}"
 }
 
 resource "aws_api_gateway_deployment" "test" {
```

Comment:
might want to have a think about whether provide multiple deployments as an option.

Reply:
Quite possibly, but later!